crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -3914,7 +3914,7 @@ impl<'db> TypeInferenceBuilder<'db> {
     fn infer_comprehension_definition(
```

Comment:
I think the support for inferring the `ExprName` targets in an unpacking is missing. Is that intentional?

I think it'd be useful to add full support for unpacking in comprehension in this PR. You can look at what kind of update is required in `infer_for_statement_definition`:

https://github.com/astral-sh/ruff/blob/b5d529e976d08e9cc03099d803db7561a4b65ebe/crates/red_knot_python_semantic/src/types/infer.rs#L3091-L3106

Reply:
I'd also be OK with doing that part (and adding tests for unpacking to Name nodes in comprehensions) as a follow-up PR. But in that case it would be good to add TODOs in this PR, just in case the follow-up doesn't happen soon.